src/modules/RootPart.rb

```diff
@@ -1718,7 +1718,13 @@ def create_backup
     # the network connection works for the chrooted scripts
     def inject_intsys_files
       # the original file is backed up and restored later
-      ::FileUtils.cp(RESOLV_CONF, File.join(Installation.destdir, RESOLV_CONF)) if File.exist?(RESOLV_CONF)
+      target = File.join(Installation.destdir, RESOLV_CONF)
+      ::FileUtils.cp(RESOLV_CONF, target) if File.exist?(RESOLV_CONF)
```

Comment:
when we are touching it we should also check if resolv.conf is not just symlink like new network manager or systemd-resolver doing and in such case also do not touch it.

Reply:
Ummm, and will then the name resolution work in the `/mnt` chroot? 

I mean when using standard wicked configration the `resolv.conf` looks like this if the system is not up:

```
### /etc/resolv.conf file autogenerated by netconfig!
#
# Before you change this file manually, consider to define the
# static DNS configuration using the following variables in the
# /etc/sysconfig/network/config file:
#     NETCONFIG_DNS_STATIC_SEARCHLIST
#     NETCONFIG_DNS_STATIC_SERVERS
#     NETCONFIG_DNS_FORWARDER
# or disable DNS configuration updates via netconfig by setting:
#     NETCONFIG_DNS_POLICY=''
#
# See also the netconfig(8) manual page and other documentation.
#
# Note: Manual change of this file disables netconfig too, but
# may get lost when this file contains comments or empty lines
# only, the netconfig settings are same with settings in this
# file and in case of a "netconfig update -f" call.
#
### Please remove (at least) this line when you modify the file!
```

So it's only an empty template and the name resolution will not work!

We need the name resolution for the registration rollback, we basically run `chroot /mnt SUSEConnect --rollback`. With empty or invalid `resolv.conf` this won't work... :worried:

Reply:
I understand when you overwrite file, but I have to be sure that when you do backup it respect symlink and also in this case that cp never write to symlink target.
try this in shell ( scenario we have to avoid ):
```
jreidinger@linux-vvcf:/tmp> touch a
jreidinger@linux-vvcf:/tmp> ln -s a b
jreidinger@linux-vvcf:/tmp> cat a
jreidinger@linux-vvcf:/tmp> cat b
jreidinger@linux-vvcf:/tmp> echo "test" > c
jreidinger@linux-vvcf:/tmp> cp c b
jreidinger@linux-vvcf:/tmp> cat a
test
jreidinger@linux-vvcf:/tmp> cat b
test
```